### ISSUE TITLE
Lack of screen reader Accessibility

### ISSUE THREAD

**Issue by TonPh2238** [DamRsn/NeuralNote#94]
Hello,
I was looking for a plugin that would be able to convert my acoustic recordings to MIDI without the extra trouble of having to replay it on a digital instrument just to get the MIDI file I need. So far Neuralnote works great, it’s incredibly accurate and fast once it’s picked up on the source file. There is a big issue though, I’m a blind user on windows with the NVDA screen reader and Reaper as my DAW. When I insert Neuralnote, only the record button is labeled and then the playback button after it has picked up on the audio I play from a different track. There seems to be no way for me to insert audio without recording playback which can be time consuming, and once that audio is recorded I don’t have access to the three knobs: “note sensability”, “split”, “min note duration”. I needed sighted help to adjust those and there doesn’t seem to be a way to save the MIDI or export it straight from the plugin window once I’m satisfied with it. I know that normally you can drag and drop items in and out of the FX window, but that is not possible for me with a screen reader and strangely the tracks seem to disappear once NeuralNote is added, although the latter may be a Reaper exclusive issue.

**Comment by DamRsn** [comment]
Hey, thanks for reaching out!

Concerning the inaccessibility for  you of the sensitivity, split and min duration knobs, as well as other parameters, there's ongoing work on making those accessible to the DAW as regular parameters. This should make those available just like the recording button. 

Regarding the import of an audio file, or the export of the midi file, I'm not sure how I can do this so that it can work easily with your setup. Currently you can drag and drop for audio import and midi export. For audio import, there's also a file chooser that opens when the audio region is clicked. Do you know any import/export technique that I could implement that would make this work well with your setup ? 

I'm currently pretty busy with work so I don't have a lot of time to work on NeuralNote, so this may take a little while. But I'm willing to make NeuralNote as accessible as I can.

**Comment by TonPh2238** [comment]
Hey there, thank you so much for responding!It’s great to hear the those parameters are being worked on. For my purposes I really have only adjusted the note sensitivity to 0 which a sighted person with no DAW knowledge can easily see and do. The additional screen reader accessibility will be great for others I know in the Reaper group who would use this for more elaborate projects.My main concern for now is importing an audio file then exporting the MIDI. When I add NeuralNote initially, I hear about 5 buttons and only two are recognized and function when I select them, those being the record and mute/unmute audio buttons. I don’t see an audio region or a button to choose a file, but I can work around this by simply recording the audio from another track. As far as exporting or saving the MIDI, my suggestion is to have a text labeled save or export MIDI button that opens the file directory to allow me to save it as though I was saving a project or track normally in the DAW. That seems to already be the case for importing audio but for some reason NVDA doesn’t see it. If it would help, I can send a screen recording of exactly what NVDA is able to view.  TonOn Mar 24, 2024, at 1:33 PM, Damien Ronssin ***@***.***> wrote:﻿
Hey, thanks for reaching out!
Concerning the inaccessibility for  you of the sensitivity, split and min duration knobs, as well as other parameters, there's ongoing work on making those accessible to the DAW as regular parameters. This should make those available just like the recording button.
Regarding the import of an audio file, or the export of the midi file, I'm not sure how I can do this so that it can work easily with your setup. Currently you can drag and drop for audio import and midi export. For audio import, there's also a file chooser that opens when the audio region is clicked. Do you know any import/export technique that I could implement that would make this work well with your setup ?
I'm currently pretty busy with work so I don't have a lot of time to work on NeuralNote, so this may take a little while. But I'm willing to make NeuralNote as accessible as I can.

**Comment by Timtam** [comment]
Hi there, just wanted to elaborate further on this:

Adding the parameters to the list of parameters accessible to the DAW will solve those issues, so thanks for that in advance.
Regarding loading audio: its already possible with the help of OCR, although not nice. Right now we need to click onto the text that says "Load a file or drag and drop one here" (or something like it), which will open a typical file open dialog which then continues to work just fine. Could you just turn that one into a typical button so that the screen reader picks it up? That would make things much easier.

And regarding export: just a button that basically copies MIDI into the clipboard so that we can paste it into the DAW would be fine already I guess.

Thanks alot!

**Comment by DamRsn** [comment]
Alright I understand. I'll add that when I will finalize the parameter PR.

**Comment by Timtam** [comment]
Hello @DamRsn , thanks for pushing the 1.0 release. I noticed that improved screen reader functionalities like the one mentioned above aren't mentioned in the changelog, did you encounter any issues that we might be able to help with, or did you forget to mention them within the changes?
I'll check out the release ASAP and report back on my findings.
Thanks again!

**Comment by DamRsn** [comment]
Hey @Timtam,

Apart from the parameters registered to the DAW change, there's no improvement regarding accessibility yet. But with the other improvements I did on the project, I should now be able to release new versions more quickly. 

Yes please, let me know if the parameters (like note sensibility, split sensibility ...) are now accessible via the screen reader. 

Concerning import of audio files and export of midi transcription, I will see how I can add regular buttons that are visible to the screen reader. One idea I'm having: I've added a "Settings" button, that shows a dropdown with only one option at the moment (`Midi Out`). If you're able to access it, I could add two more options:
- Load audio file
- Export transcription to midi file
Both options would open a file explorer window.

Let me know if this would work for you.

Thank you!
